src/renderers/shared/shared/ReactTreeTraversal.js

```diff
@@ -90,13 +91,54 @@ function getParentInstance(inst) {
 /**
  * Simulates the traversal of a two-phase, capture/bubble event dispatch.
  */
+
+function isInteractive(inst) {
+  var tag = ReactDOMFeatureFlags.useFiber ? inst.type : inst._tag;
+
+  return (
+    tag === 'button' || tag === 'input' ||
+    tag === 'select' || tag === 'textarea' ||
+    tag === 'fieldset'
+  );
+}
+
+function shouldIgnoreElement(inst) {
+  if (inst && isInteractive(inst)) {
+    if (ReactDOMFeatureFlags.useFiber) {
```

Comment:
Sorry... tunnel vision. There's also the condition above it:

https://github.com/facebook/react/blob/master/src/renderers/shared/shared/ReactTreeTraversal.js#L17

```javascript
  if (inst._hostParent !== undefined) {
    return inst._hostParent;
  }
```

Which could be subbed out for `inst._currentElement` in this case.